ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/v2.scala

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state
+
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.daml.lf.transaction.{GenTransaction, Transaction}
+import com.digitalasset.daml.lf.value.Value
+
+/** Interfaces to read from and write to an (abstract) participant state.
+  *
+  * A DAML ledger participant is code that allows to actively participate in
+  * the evolution of a shared DAML ledger. Each such participant maintains a
+  * particular view onto the state of the DAML ledger. We call this view the
+  * participant state.
+  *
+  * Actual implementations of a DAML ledger participant will likely maintain
+  * more state than what is exposed through the interfaces in this package,
+  * which is why we talk about an abstract participant state. It abstracts
+  * over the different implementations of DAML ledger participants.
+  *
+  * The interfaces are optimized for easy implementation. The
+  * [[v1.WriteService]] interface contains the methods for changing the
+  * participant state (and potentially the state of the DAML ledger), which
+  * all ledger participants must support. These methods are for example
+  * exposed via the DAML Ledger API. Actual ledger participant implementations
+  * likely support more implementation-specific methods. They are however not
+  * exposed via the DAML Ledger API. The [[v1.ReadService]] interface contains
+  * the one method [[v1.ReadService.stateUpdates]] to read the state of a ledger
+  * participant. It represents the participant state as a stream of
+  * [[v1.Update]]s to an initial participant state. The typical consumer of this
+  * method is a class that subscribes to this stream of [[v1.Update]]s and
+  * reconstructs (a view of) the actual participant state. See the comments
+  * on [[v1.Update]] and [[v1.ReadService.stateUpdates]] for details about the kind
+  * of updates and the guarantees given to consumers of the stream of
+  * [[v1.Update]]s.
+  *
+  * We provide a reference implementation of a participant state in
+  * [[com.daml.ledger.participant.state.v2.impl.reference.Ledger]]. There we
+  * model an in-memory ledger, which has by construction a single participant,
+  * which hosts all parties. See its comments for details on how that is done,
+  * and how its implementation can be used as a blueprint for implementing
+  * your own participant state.
+  *
+  * We do expect the interfaces provided in
```

Comment:
Remove this explanation here; and replace it with an explanation at the very top which explains the relation between the `v2` and `v1` version of the service. 

- v1 is kept as is to serve the needs of `//ledger/api-server-damlonx`
- v2 is in-flux to evolve to serve the needs of merging the Sandbox ledger API server with the DAML on X API server as per #1273 .